src/frontend/src/catalog/root_catalog.rs

```diff
@@ -253,12 +253,24 @@ impl Catalog {
     }
 
     pub fn update_table(&mut self, proto: &PbTable) {
-        let table = self
-            .get_database_mut(proto.database_id)
-            .unwrap()
-            .get_schema_mut(proto.schema_id)
-            .unwrap()
-            .update_table(proto);
+        let database = self.get_database_mut(proto.database_id).unwrap();
+        let schema = database.get_schema_mut(proto.schema_id).unwrap();
+        let table = if schema.get_table_by_id(&proto.id.into()).is_some() {
+            schema.update_table(proto)
+        } else {
+            // Enter this branch when schema is changed by `ALTER ... SET SCHEMA ...` statement.
+            let new_table = schema.create_table(proto);
+            database
+                .iter_schemas_mut()
+                .find(|schema| {
+                    schema.id() != proto.schema_id
```

Comment:
What if the target new schema is the original one? And I'm afraid checking and ignoring it only in frontend is not enough. Frontend can also receive a update including same target schema as original in multi-frontend env. We'd better try to create it after the old catalog get dropped.

Reply:
Do you mean `ALTER TABLE tb SET SCHEMA new_schema` and `ALTER TABLE tb OWNER TO new_user` are issued at the same time? IIUC, `Meta` uses lock to ensure the sequential exec of these 2 statements, while `Frontend` handles notification from `Meta` concurrently.  Let's take `ALTER TABLE tb SET SCHEMA new_schema` as `S1`, `ALTER TABLE tb OWNER TO new_user` as `S2`.

### The case of inconsistency if *create before drop*:
Meta: `S1 -> S2`
Frontend:
|S1|S2|
|:-:|:-:|
||Alter Owner|
|Create Table||
|Drop Table||

`Alter Owner` finds there is no corresponding table in the schema because `Create Table` hasn't been executed yet. So it enters the `SET SCHEMA` branch instead of the `Update` branch.

### The case of inconsistency if *drop before create*:
Meta: `S1 -> S2`
Frontend:
|S1|S2|
|:-:|:-:|
|Drop Table||
||Alter Owner|
|Create Table||

|S1|S2|
|:-:|:-:|
||Alter Owner|
|Drop Table||
|Create Table||

As long as `Alter Owner` is executed before `Create Table`, `Alter Owner` will always enter `SET SCHEMA` branch and the inconsistency won't eliminated.

### Conclusion
The order of `Create Table` and `Drop Table` is not critical. And I can think of 2 solutions:
1. `Meta` waits for responses from all `Frontend`s. Hence the notification handler is executed sequentially in `Frontend`.
2. Lock in notification handler in `Frontend`, and some mechanism to preserve the notification order from `Meta` (maybe version).

Reply:
Sorry for my unclear comment. I mean that if a table `t1` is in schema `public`, and when we run sql like `ALTER TABLE t1 SET SCHEMA public` trying to change its schema to `public`, unwrap in L269 will panic the frontend.

And for the case you mentioned here, the notification order is already ensured by meta, which is protected by the catalog lock. As you can see in the notification manager, the notification will be queued with an associated version. So I think you don't have to worry about it.